src/apps/basket/views.py

```diff
@@ -6,22 +7,54 @@
 
 
 def basket_add(request, product_id):
+    """
+    This view function is responsible for adding a product to the user's basket.
+    It handles the form submission, validates the form data, and checks product
+    availability in stock before adding it to the basket.
+
+    -If the product is successfully added to the basket, a success message is
+    displayed to the user.
+    -If the quantity exceeds the available stock, an error message is shown.
+    -Redirects the user back to the previous page.
+    """
     basket = Basket(request)
     product = get_object_or_404(Product, id=product_id)
     form = BasketAddProductForm(request.POST)
+
     if form.is_valid():
         cd = form.cleaned_data
-        basket.add(product=product, quantity=cd["quantity"], update_quantity=cd["update"])
+
+        quantity_in_basket = basket.get_quantity(product)
+        total_quantity = quantity_in_basket + cd["quantity"]
+
+        if product.stock < cd["quantity"] or product.stock < total_quantity:
+            messages.error(
+                request, f"Insufficient quantity of goods in stock. Available {product.stock}"
+            )
```

Comment:
Imagine the situation. We have product with current stock = 10. 
User has in basket already 3. He goes to page and see again 10 (and it's fine, as order has not been completed)
User tries to add 9 more to basket. Now error appears. It's fine. 
But text the user see: Insufficient quantity of goods in stock. Available 10 (but he just tried to added 9). It will be confusing I guess

Reply:
Good comment! I thought about different situations yesterday and I believe added this text when i tried to reduce quantity after adding into the basket. I will think it over a bit and fix.